### ISSUE TITLE
Android apps crash when attaching

### ISSUE THREAD

**Issue by rumpeltux** [frida/frida#374]
Following the tutorial, `frida-ps -U` works fine, but `frida-trace -U -i open com.android.chrome` crashes Chrome and frida responds:

    Failed to attach: timed out while waiting for session to establish

This is on a Pixel 2, running Android 8.
Frida client/server are v10.6.21.
From `adb logcat`:
```
11-06 09:52:37.788 17014 17240 F libc    : Fatal signal 11 (SIGSEGV), code 2, fault addr 0xf5148038 in tid 17240 (.android.chrome), pid 17014 (.android.chrome)

11-06 09:52:37.822 17258 17258 F DEBUG   : pid: 17014, tid: 17240, name: .android.chrome  >>> com.android.chrome <<<
11-06 09:52:37.822 17258 17258 F DEBUG   : signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0xf5148038
11-06 09:52:37.822 17258 17258 F DEBUG   :     r0 ec1f2360  r1 00000001  r2 00000001  r3 00000000
11-06 09:52:37.822 17258 17258 F DEBUG   :     r4 ce67f970  r5 ce67f970  r6 ce67f970  r7 0000003d
11-06 09:52:37.822 17258 17258 F DEBUG   :     r8 00004276  r9 00004276  sl ec1f2081  fp 0000000b
11-06 09:52:37.822 17258 17258 F DEBUG   :     ip 0000003d  sp ce67f948  lr ec1f20a5  pc f5148038  cpsr 000b0030
11-06 09:52:37.823 17258 17258 F DEBUG   : 
11-06 09:52:37.823 17258 17258 F DEBUG   : backtrace:
11-06 09:52:37.823 17258 17258 F DEBUG   :     #00 pc 00002038  /dev/__properties__/u:object_r:log_tag_prop:s0
```

**Comment by rumpeltux** [comment]
I experience similar behavior in the emulator on x86.
With earlier api-levels native tracing works, although I can’t get any of the java examples to run, e.g. the onResume example in https://www.frida.re/docs/javascript-api/#java always crashes the app (since super.onResume() wasn’t called) and never logs anything.
Is there a list of supported platforms / APIs?

**Comment by lupodellemontagne** [comment]
I have the same behaviour on Mac Os X 10.12.6. Using the android emulator shipped with android studio and Android 6.0 with x86 CPU. The app crashes both using the onResume hooking and also if I try to enumerates all objects of a certain instance on the heap, using the script:

`console.log("Script loaded successfully ");
Java.perform(function () {
        Java.choose("com.opera.install.ConfReader",{
          onMatch: function (instance) {
                    console.log(""+instance);
                 }
        ,
        onComplete: function () {
                 }
        }
        );
});` 

The server version of frida is frida-server-10.6.21-android-x86.
The app that I am trying to hook is a really simple malware that has not any obfuscation or anti-debugger capabilities:
https://koodous.com/apks/58687779d3d4455e0e0e3b5ed849b4bcd10d2909a01172e4b3981ba30fb66719/analysis

I tried several version of frida server, and using frida-server-10.6.21-arm with the emulation of android kitkat 4.4 arm, the hooks work even if in a really unreliable way. A lot of crashes.

**Comment by oleavr** [comment]
Fixed in 10.6.40.
